01/implement-most-frequent-k-elements.checkpoint.md

```diff
@@ -0,0 +1,238 @@
+# Most Frequent K Elements
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: c371a9d2-49b1-417b-abc4-3765e5cade18
+* title: Most Frequent K Elements
+* points: 3
+### !question
+
+Create a function, `most_frequent_k_elements`, that takes a non-empty array of integers and return the k most frequent elements. (Note: You may assume k is always valid, 1 ≤ k ≤ number of unique elements.)
+
+**Example 1:**
+
+```
+Input: nums = [1,1,1,2,2,3], k = 2
+Output: [1,2]
+```
+
+**Example 2:**
+```
+Input: nums = [1], k = 1
+Output: [1]
+```
+
+### !end-question
+### !placeholder
+
+```python
+def most_frequent_k_elements(num_rows):
+    """
+    A function that returns the first num_rows of Pascal's triangle.
+  
+    Parameters:
+    arr (array): a non-empty array of integers
+    k (int): number of most frequent integers to return 
+  
+    Returns:
+    array[int]: 1D array that that returns the k most frequent elements
+    """
+    pass
+```
+### !end-placeholder
+### !tests
+```python
+import unittest
+from main import *
+
+class TestChallenge(unittest.TestCase):
+    def test_most_k_frequent_elements_nominal(self):
+        # Arrange
+        arr = [1,1,1,2,2,3]
+        k = 2
+
+        # Act
+        result = most_frequents_k_elements(arr, k)
+
+        # Assert
+        self.assertEqual(result, [1,2])
```

Comment:
Since no explicit result ordering was specified, can we do order-independent testing for the tests?

Reply:
✅

Reply:
By order-independent, I imagined wrapping the result list in a set and comparing to an expected set, so that it doesn't matter whether the submitted implementation puts the results in order of most to least frequent, since order wasn't a stated requirement.

Reply:
OH. I understand now!

Reply:
✅